### PR TITLE
add jenkins api

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

**Comment by soulseen** [comment]
@runzexia @zryfish @wansir

File: pkg/apiserver/devops/devops.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	projectName := req.PathParameter("projectName")
+	pipelineName := req.PathParameter("pipelineName")
+	branchName := req.PathParameter("branchName")
+	runId := req.PathParameter("runId")
+	nodeId := req.PathParameter("nodeId")
+	stepId := req.PathParameter("stepId")
+
+	res, err := devops.GetStepLog(projectName, pipelineName, branchName, runId, nodeId, stepId, req.Request)
+	if err != nil {
+		parseErr(err, resp)
```
Comment (zryfish) [inline pkg/apiserver/devops/devops.go]: log error

File: pkg/models/devops/devops.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	return res, err
+}
+
+func SearchPipelines(req *http.Request) ([]interface{}, error) {
+	baseUrl := JenkinsUrl + SearchPipelineUrl + req.URL.RawQuery
+	log.Infof("Jenkins-url: " + baseUrl)
+
+	resBody, err := jenkinsClient(baseUrl, req)
+	if err != nil {
+		return nil, err
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: same here

File: pkg/apis/devops/v1alpha2/register.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		Param(webservice.PathParameter("projectName", "devops project name")).
+		Param(webservice.PathParameter("pipelineName", "pipeline name")).
+		Param(webservice.PathParameter("branchName", "pipeline branch name")).
+		Param(webservice.PathParameter("runId", "pipeline runs id")).
+		Param(webservice.PathParameter("nodeId", "pipeline runs node id")).
+		Param(webservice.PathParameter("stepId", "pipeline runs step id")).
+		Param(webservice.QueryParameter("start", "start").
+			Required(true).
+			DataFormat("start=%d").
+			DefaultValue("start=0")))
```
Comment (zryfish) [inline pkg/apis/devops/v1alpha2/register.go]: should dataformat applied just for value? I think default value is `0`, not `start=0`

File: pkg/models/devops/devops.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	return res, err
+}
+
+func GetStepLog(projectName, pipelineName, branchName, runId, nodeId, stepId string, req *http.Request) ([]byte, error) {
+	baseUrl := fmt.Sprintf(JenkinsUrl+GetStepLogUrl+req.URL.RawQuery, projectName, pipelineName, branchName, runId, nodeId, stepId)
+	log.Infof("Jenkins-url: " + baseUrl)
+
+	resBody, err := jenkinsClient(baseUrl, req)
+	if err != nil {
+		return nil, err
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: log error

File: pkg/apis/devops/v1alpha2/register.go (hunk 3 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: like this: 
```
{
            "type": "string",
            "format": "start=%d",
            "default": "start=0",
            "description": "start",
            "name": "start",
            "in": "query",
            "required": true
}
```
I think `start=0` is more completely and clearly. The previous Document is also this format。

Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: The key is fixed. I think no need to format the key.

**Pull request merged**